File: discover_javascript/write_them_to_disc.js
var https = require('https');

var options = {
    hostname: 'api.github.com',
    path: '/search/repositories?q=language:javascript&sort=stars&order=desc',
    headers: {
        'User-Agent': 'Holberton_School',
        'Authorization': 'token ' + process.env.TOKEN
    }
};

var req = https.request(options, function(res) {
    var fs = require('fs');
    fs.writeFile("/tmp/15", ,function(err) {
        if(err) {
            return console.log(err);
        }

        console.log("The file was saved!");
    });
});
req.end();

req.on('error', function(e) {
    console.error(e);
});

function streamToString(stream, cb) {
    const chunks = [];
    stream.on('data', (chunk) => {
        chunks.push(chunk);
    });
    stream.on('end', () => {
        cb(chunks.join(''));
    });
}

File: discover_more_python/write_them_to_disc.py
import requests 
r = requests.get('https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc', headers={
    'User-Agent': 'Holberton_School',
    'Authorization': 'token c1485ae8f00e64f0aa058a33db2554cdd8052cd6'
})
with open('/tmp/15', 'w') as f:
    f.write(str(r.json()))
print ("The file was saved!")
